lib/philomena_web/post_json.ex

```diff
@@ -1,13 +1,34 @@
 defmodule PhilomenaWeb.PostJson do
   alias PhilomenaWeb.UserAttributionView
 
+  def as_json(%{topic: %{hidden_from_users: true}} = post) do
+    %{
+      id: post.id,
+      topic_id: post.topic_id,
+      user_id: nil,
+      author: nil,
+      body: nil
+    }
+  end
```

Comment:
Style: function clauses of the same name and arity should have no blank line between them